### ISSUE TITLE
Create a plug-in API to make it easier to modify or improve MFEs without forking

### ISSUE THREAD

**Issue by jmakowski1123** [openedx/axim-engineering#430]
### Problem

edx-platform already has plugins, and it has made development much better.  Some form of extensibility architecture should be created so that MFEs can be modified or improved without the necessity to fork them.

### Product/Platform Value

Like runtime configuration, a solution for customization would allow use-cases that are not currently possible with MFEs but are possible with Comprehensive Theming.  Feature parity here would greatly improve adoption, as it would allow many providers to actually upgrade to newer releases.

### Related or in-progress work

- [x] https://github.com/openedx/tcril-engineering/issues/521

### Contingencies

**Comment by brian-smith-tcril** [comment]
I [left a comment](https://github.com/openedx/tcril-engineering/issues/429#issuecomment-1280996389) related to this on https://github.com/openedx/tcril-engineering/issues/429

tl;dr - what are people doing with comprehensive theming that we don't support in MFEs? can we support that by making paragon components more customizable? what should the dev/deployment flows for plugins look like?

**Comment by arbrandes** [comment]
@brian-smith-tcril 

> what are people doing with comprehensive theming that we don't support in MFEs?

"Comprehensive theming" is essentially overriding templates server-side.  That allows for overriding basically _anything_ in a page.  People would (ab)use this to their heart's content.

When moving to micro-frontends, we lost this ability, and will likely never get it back completely.  The motivation here is precisely to find out how we can get to a place where 80% of the customization needs (including theming/branding) are covered without requiring users to fork the MFEs.

**Comment by arbrandes** [comment]
This topic was discussed in today's Architecture Coordination Working Group [meeting](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3562930177/2022-10-26+Arch+Coordination+Meeting+notes).

Looks like we might have/want to move this up.

**Comment by arbrandes** [comment]
Found [this](https://github.com/openedx/frontend-app-learning/pull/581): a specific use case for plugins that I had not seen, yet.

**Comment by arbrandes** [comment]
This is technically the primary roadmap issue behind the upcoming [Frontend Pluggability Summit](https://discuss.openedx.org/t/frontend-pluggability-summit/11167/9).  This is the summit's [current wiki page](https://openedx.atlassian.net/l/cp/sdRpN2Tw).

**Comment by arbrandes** [comment]
Actually... This is the main roadmap issue:

https://github.com/openedx/platform-roadmap/issues/27

I'm closing this one to avoid confusion.
